src/LibraryManager.Vsix/UI/InstallDialogTestContract.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Threading;
+
+namespace Microsoft.Web.LibraryManager.Vsix.UI
+{
+    /// <summary>
+    /// This class gives apex access to opened - add client side libraries dialog.
+    /// </summary>
+    internal class InstallDialogTestContract
+    {
+        public static IInstallDialogTestContract window;
+        // This event lets apex know when the dialog is open.
+        public static EventWaitHandle windowIsUp = new EventWaitHandle(false, EventResetMode.ManualReset);
```

Comment:
>EventWaitHandle [](start = 22, length = 15)

Curious if this would look nicer with a TaskCompletionSource instead

Reply:
I couldn't get this to work with TaskCompletionSource and don't fully understand the benefits if we switched over. I'll keep this open and sync up with you in person.

---
In reply to: [205456459](https://github.com/aspnet/LibraryManager/pull/332#discussion_r205456459) [](ancestors = 205456459)

Reply:
I guess, for me, EventWaitHandle adds yet another concept to understand. As I need to understand Tasks already, it felt like we should use that concept if possible.

---
In reply to: [205896628](https://github.com/aspnet/LibraryManager/pull/332#discussion_r205896628) [](ancestors = 205896628,205456459)

Reply:
Updating this in iteration 4. 

---
In reply to: [205906211](https://github.com/aspnet/LibraryManager/pull/332#discussion_r205906211) [](ancestors = 205906211,205896628,205456459)

Reply:
Something still doesn't feel right. Will start a new comment.

---
In reply to: [205919145](https://github.com/aspnet/LibraryManager/pull/332#discussion_r205919145) [](ancestors = 205919145,205906211,205896628,205456459)